### PR TITLE
Make team registration form for breakin

### PR DESCRIPTION
**Pull request by hharchani**
Someone needs to test this please!

### CODE REVIEW

File: src/app/controllers/contest.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        if (!$team_info && $_SERVER['REQUEST_METHOD'] === 'POST') {
+            $team_name = null;
+            $teammate_nick = null;
+            if (isset($_POST['teammate_nick'])) {
+                if (!isset($_POST['team']) || !trim($_POST['team'])) {
+                    $errors['team'] = "Please enter a valid team name";
+                }
+                $this->load_model('auth_model');
+                if ($_POST['teammate_nick'] == $user_nick) {
+                    $errors['teammate_nick'] = "<pre>¯\_(ツ)_/¯</pre>";
```
Comment (ParthKolekar) [inline src/app/controllers/contest.php:22]:
```
¯\_(ツ)_/¯
```

**Comment by ParthKolekar** [comment]
LGTM

:+1:

File: src/app/controllers/contest.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                    } else {
+                        $errors['teammate_nick'] = "Please enter a valid <strong>registered user's</strong> nick";
+                    }
+                }
+            }
+            if (isset($_POST['team']) && trim($_POST['team'])) {
+                if ($this->breakin_model->does_team_exists($_POST['team'])) {
+                    $errors['team'] = "This team name is already taken";
+                } else {
+                    $team_name = trim($_POST['team']);
```
Comment (ParthKolekar) [inline src/app/controllers/contest.php:47]: Remove unicode characters support.

File: src/app/controllers/contest.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+        $this->load_view('skeleton_template/header', [
+            'is_authenticated'  => true,
+            'user_nick'         => $user_nick,
+        ]);
+
+        $this->load_model('breakin_model');
+        $team_info = $this->breakin_model->get_team_info($user_nick);
+        $errors = [];
+        if (!$team_info && $_SERVER['REQUEST_METHOD'] === 'POST') {
```
Comment (ParthKolekar) [inline src/app/controllers/contest.php:16]: Django: 
    @require_http_methods(["POST"]) 

:grin:

File: src/app/routes.php [hunk 4]
```diff
@@ -24,6 +24,8 @@
     "/sap/portal/"                      => "/sap_portal/",
     "/sap/"                             => "/sap/",
 
+    "/threads/breakin/register"         => "/contest/breakin/",
```
Comment (nisargjhaveri) [inline src/app/routes.php]: Trailing slash?

**Pull request merged**